lib/vrl/stdlib/src/parse_ruby_hash.rs

```diff
@@ -109,50 +140,88 @@ fn parse_nil<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, Val
 fn parse_bytes<'a, E: HashParseError<&'a str>>(input: &'a str) -> IResult<&'a str, Bytes, E> {
     context(
         "bytes",
-        map(
-            preceded(char('\"'), cut(terminated(parse_str::<'a, E>, char('\"')))),
-            |value| Bytes::copy_from_slice(value.as_bytes()),
-        ),
+        map(alt((parse_str('"'), parse_str('\''))), |value| {
+            Bytes::copy_from_slice(value.as_bytes())
+        }),
     )(input)
 }
 
-fn parse_key<'a, E: HashParseError<&'a str>>(input: &'a str) -> IResult<&'a str, String, E> {
-    context(
-        "string",
-        map(
-            alt((
-                preceded(char('\"'), cut(terminated(parse_str, char('\"')))),
-                preceded(char('\''), cut(terminated(parse_str, char('\'')))),
-                alphanumeric,
-            )),
-            String::from,
-        ),
+fn parse_symbol_key<T, E: ParseError<T>>(input: T) -> IResult<T, T, E>
+where
+    T: std::fmt::Display,
+    T: InputTakeAtPosition,
+    <T as InputTakeAtPosition>::Item: AsChar,
+{
+    take_while1(move |item: <T as InputTakeAtPosition>::Item| {
+        let c = item.as_char();
+        c.is_alphanum() || c == '_'
+    })(input)
+}
+
+fn parse_colon_key<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, &'a str, E> {
+    map(consumed(preceded(char(':'), parse_symbol_key)), |res| res.0)(input)
+}
+
+fn parse_key_arrow_hash<'a, E: HashParseError<&'a str>>(
```

Comment:
Also known as the "[hash rocket](https://web.archive.org/web/20130909062027/http://peepcode.com/blog/2011/rip-ruby-hash-rocket-syntax)".

Reply:
From where I come from it's `fat arrow` 🤷‍♂️